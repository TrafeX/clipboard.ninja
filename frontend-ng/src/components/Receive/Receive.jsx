import {
  Box,
  Code,
  Flex,
  List,
  ListIcon,
  ListItem
} from "@chakra-ui/react";
import { IoReturnDownForwardSharp } from "react-icons/io5";

const Receive = ({ receivedMessages }) => (
  <Box>
    <List>
      {receivedMessages &&
        receivedMessages.map((message, i) => {
          return (
            <ListItem key={i}>
              <Flex direction="row">
                <ListIcon as={IoReturnDownForwardSharp} color={"brand.600"} />
                <Code style={{whiteSpace: "pre-wrap"}}>{message}</Code>
              </Flex>
            </ListItem>
          );
        })}
    </List>
  </Box>
);

export default Receive;
