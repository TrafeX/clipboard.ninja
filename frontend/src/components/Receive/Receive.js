// @flow
import React from 'react';

const style = {
  ul: {
    textAlign: 'left',
  }
};

type Props = {
  receivedMessages: Array<string>,
}

const Receive = ({receivedMessages}: Props) => (
  <div>
    <ul style={style.ul}>
    { receivedMessages && receivedMessages.map((message, i) => {
      return <li key={i}><pre>{message}</pre></li>
    }) }
    </ul>
  </div>
);

export default Receive;
