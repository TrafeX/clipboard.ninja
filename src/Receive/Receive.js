// @flow
import React from 'react';

const style = {
  ul: {
    textAlign: 'left',
  }
};

type Props = {
  receivedMessages: Array<string>,
}

const Receive = (props: Props) => (
  <div>
    <ul style={style.ul}>
    { props.receivedMessages.map((message, i) => {
      return <li key={i}><pre>{message}</pre></li>
    }) }
    </ul>
  </div>
);

export default Receive;
