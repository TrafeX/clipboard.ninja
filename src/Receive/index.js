import Receive from './Receive';

export default Receive;
