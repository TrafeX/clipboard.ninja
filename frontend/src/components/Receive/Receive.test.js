import React from 'react';
import Receive from './Receive';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const wrapper = shallow(<Receive receivedMessages={['Hello']} />);
  expect(wrapper).toMatchSnapshot();
});
