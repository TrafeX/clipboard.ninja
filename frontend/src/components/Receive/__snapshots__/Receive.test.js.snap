// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <ul
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <li
      key="0"
    >
      <pre>
        Hello
      </pre>
    </li>
  </ul>
</div>
`;
